File: C++/151. Reverse Words in a String.cpp
class Solution {
public:
    string reverseWords(string s) {
        string ans = "";
        string w = "";
        int i = 0;
        while(i < s.size()){
            while(i < s.size() && s[i] != ' '){
                w += s[i];
                i++;
            }
            if (w.size() != 0) {
                if (ans.size() != 0) {
                    ans = w + " " + ans;
                }
                else {
                    ans = w;
                }
                w = "";
            }
            i++;
        }
        return ans;
    }
};

File: JavaScript/151. Reverse Words in a String.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    var ans = "";
    var w = "";
    var i = 0;
    while(i < s.length){
        while(i < s.length && s[i] != ' '){
            w += s[i];
            i++;
        }
        if (w.length != 0) {
            if (ans.length != 0) {
                ans = w + " " + ans;
            }
            else {
                ans = w;
            }
            w = "";
        }
        i++;
    }
    return ans;
};
